src/Notes/WC_Admin_Notes_WooCommerce_Payments.php

```diff
@@ -0,0 +1,145 @@
+
+	/**
+	 * Attach hooks.
+	 */
+	public function __construct() {
+
+		add_action( 'woocommerce_note_action_install-now', array( $this, 'install' ) );
+
+		add_action( 'wc-admin-woocommerce-payments_add_note', array( $this, 'add_note' ) );
+	}
+
+	/**
+	 * Maybe add a note on WooCommerce Payments for US based sites older than a week without the plugin installed.
+	 */
+	public static function possibly_add_note() {
+
+		if ( ! self::wc_admin_active_for( WEEK_IN_SECONDS ) || 'US' !== WC()->countries->get_base_country() ) {
+			return;
+		}
+
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// We already have this note? Then exit, we're done.
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+		if ( ! empty( $note_ids ) ) {
+
+			$note_id = array_pop( $note_ids );
+			$note    = WC_Admin_Notes::get_note( $note_id );
+
+			// If the WooCommerce Payments plugin was installed after the note was created, make sure it's marked as actioned.
+			if ( self::validate_plugin() && WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED !== $note->get_status() ) {
+				$note->set_status( WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED );
+				$note->save();
+			}
+
+			return;
+		}
+
+		$current_date = new \DateTime();
+		$publish_date = new \DateTime( '2020-04-14' );
+
+		if ( $current_date >= $publish_date ) {
+
+			self::add_note();
+
+		} else {
+
+			$hook_name = sprintf( '%s_add_note', self::NOTE_NAME );
+
+			if ( ! WC()->queue()->get_next( $hook_name ) ) {
+				WC()->queue()->schedule_single( $publish_date->getTimestamp(), $hook_name );
+			}
+		}
+	}
+
+	/**
+	 * Add a note about WooCommerce Payments.
+	 */
+	public static function add_note() {
+
+		$note = new WC_Admin_Note();
+		$note->set_title( __( 'Try the new way to get paid', 'woocommerce-admin' ) );
+		$note->set_content( __( 'Securely accept credit and debit cards on your site. Manage transactions without leaving your WordPress dashboard. Only with WooCommerce Payments.', 'woocommerce-admin' ) );
+		$note->set_content_data( (object) array() );
+		$note->set_type( WC_Admin_Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_icon( 'credit-card' );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_source( 'woocommerce-admin' );
+		$note->add_action( 'learn-more', __( 'Learn more', 'woocommerce-admin' ), 'https://woocommerce.com/payments/', WC_Admin_Note::E_WC_ADMIN_NOTE_UNACTIONED );
+		$note->add_action( 'install-now', __( 'Install now', 'woocommerce-admin' ), false, WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED, true );
+
+		// Create the note as "actioned" if the plugin is already installed.
+		if ( self::validate_plugin() ) {
+			$note->set_status( WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED );
+		}
+
+		$note->save();
+	}
+
+
+	/**
+	 * Add a note on WooCommerce Payments.
+	 */
+	protected static function validate_plugin() {
+		include_once ABSPATH . '/wp-admin/includes/plugin.php';
+		return 0 === validate_plugin( self::PLUGIN_FILE );
```

Comment:
While likely an edge case, if the plugin is installed in a differently named directory, this check might return false if wcpay is installed. Is there a constant, or a method name distinct to wcpay that we can use to sniff out if it is installed with instead?

Reply:
@timmyc thanks for the review. I've added SHA: 96e9e7d4 to check against the main `WC_Payments` class in addition to `validate_plugin()` (I also renamed the method given the double check and to better indicate it returns a boolean).

The method still checks against `validate_plugin( self::PLUGIN_FILE )` if the constant isn't defined, because we want to mark the notification as actioned even if the plugin has been installed and subsequently deactivated before `'wc_admin_daily'` is run again.